nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/FlowResource.java

```diff
@@ -442,6 +447,26 @@ public Response getFlowMetrics(
             return generateOkResponse(response)
                     .type(MediaType.TEXT_PLAIN_TYPE)
                     .build();
+        } else if ("json".equalsIgnoreCase(producer)) {
+            final Collection<CollectorRegistry> allRegistries = serviceFacade.generateFlowMetrics();
+            final Map<String, List<Collector.MetricFamilySamples.Sample>> response = new HashMap<>();
```

Comment:
Did you consider writing out the response using `StreamingOutput` analogous to the current Prometheus approach?